libs/api_common/windows_platform_common.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#include <cassert>
+#include <stdexcept>
+#include "api_internal.h"
+#include "api_common.hpp"
+#pragma warning(push)
+#pragma warning(disable : 4100) // 'identifier' : unreferenced formal parameter
+#pragma warning(disable : 4244) // 'conversion' conversion from 'type1' to
+                                // 'type2', possible loss of data
+#include "crab_verifier.hpp"
+#pragma warning(pop)
+#include "ebpf_api.h"
+#undef VOID
+#include "ebpf_helpers.h"
+#include "helpers.hpp"
+#include "map_descriptors.hpp"
+#include "platform.hpp"
+#include "spec_type_descriptors.hpp"
+#include "windows_platform.hpp"
+
+#define PTYPE(name, descr, native_type, prefixes) \
+    {                                             \
+        name, descr, native_type, prefixes        \
+    }
+
+#define PTYPE_PRIVILEGED(name, descr, native_type, prefixes) \
+    {                                                        \
+        name, descr, native_type, prefixes, true             \
+    }
+
+// Allow for comma as a separator between multiple prefixes, to make
+// the preprocessor treat a prefix list as one macro argument.
+#define COMMA ,
+
+const EbpfContextDescriptor g_xdp_context_descriptor = {
+    24, // Size of ctx struct.
+    0,  // Offset into ctx struct of pointer to data, or -1 if none.
+    8,  // Offset into ctx struct of pointer to end of data, or -1 if none.
+    16, // Offset into ctx struct of pointer to metadata, or -1 if none.
+};
+
+const EbpfProgramType windows_xdp_program_type =
+    PTYPE("xdp", &g_xdp_context_descriptor, (uint64_t)&EBPF_PROGRAM_TYPE_XDP, {"xdp"});
+
+const EbpfContextDescriptor g_bind_context_descriptor = {
+    43, // Size of ctx struct.
+    0,  // Offset into ctx struct of pointer to data, or -1 if none.
+    8,  // Offset into ctx struct of pointer to end of data, or -1 if none.
+    -1, // Offset into ctx struct of pointer to metadata, or -1 if none.
+};
+
+const EbpfProgramType windows_bind_program_type =
+    PTYPE("bind", &g_bind_context_descriptor, (uint64_t)&EBPF_PROGRAM_TYPE_BIND, {"bind"});
+
+const std::vector<EbpfProgramType> windows_program_types = {
+    PTYPE("unspecified", {0}, 0, {}),
+    windows_xdp_program_type,
+    windows_bind_program_type,
+};
+
+EbpfProgramType
+get_program_type_windows(const GUID& program_type)
+{
+    // TODO: (Issue #205) Make an IOCTL call to fetch the program context
```

Comment:
Per triage meeting, 205 is a duplicate of 67

Reply:
Updated comment with the issue#